system-addon/test/unit/content-src/components/TopSites.test.jsx

```diff
@@ -438,3 +450,158 @@ describe("<TopSitesEdit>", () => {
     links.forEach((link, i) => assert.equal(link.props().link.url, rows[i].url));
   });
 });
+
+describe("<TopSiteForm>", () => {
+  let wrapper;
+  let sandbox;
+
+  function setup(props = {}) {
+    sandbox = sinon.sandbox.create();
+    const customProps = Object.assign({}, {onClose: sandbox.spy(), dispatch: sandbox.spy()}, props);
```

Comment:
I'm not sure dispatch should be a spy. Having events actually being dispatched could cause additional side effects that might not be properly handled by tests (I'm thinking here that the test will pass accidentally). The proper way (my2c) would be to make it a stub, test that it gets called when specific buttons are clicked and for the effect of those clicks we should manually emit the correct events in a separate test and assert the results. I'm not going to hold the patch on this, just something for the future. Unfortunately we have a lot of weird issues so I think we should be stricter with our tests.

Reply:
I agree. I think that would be more like integration testing as opposed to unit testing? We probably need more tests that load all components and make sure they all place nicely and don't affect each other. For these particular tests though, I'm not sure what difference a spy vs stub would make since it's super controlled and the component gets loaded in isolation. I'm open to working on making it all better though.